src/main/java/luke/Parser.java

```diff
@@ -0,0 +1,142 @@
+package luke;//Returns a list of Strings neatly parsed as arguments.
+import java.io.File;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class Parser {
+    private File saveFile;
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Creates a parser with the specified Storage and save file.
+     *
+     * @param storage Serializable which represents the save data.
+     * @param saveFile File object used to save/load.
+     */
+    public Parser(Storage storage, File saveFile) {
+        this.storage = storage;
+        ArrayList<Task> tasks = storage.getHistory();
+        this.taskList = new TaskList(tasks);
+        this.saveFile = saveFile;
+    }
+
+    /**
+     * Parses a command given a string input. Returns false if there is a next command, else true.
+     *
+     * @param input String input as given by the user.
+     * @return False if there is a next command, else true.
+     */
+    public boolean parseCommand(String input) {
+        String trimmedLowercase = input.trim().toLowerCase();
+        if (trimmedLowercase.equals("bye")) {
```

Comment:
Maybe consider switching to a case statement rather than relying on if-else constructs, it could potentially enhance code readability.